common/remote/rmtspawn.cpp

```diff
@@ -140,7 +140,13 @@ ISocket * spawnRemoteChild(SpawnKind kind, const char * exe, const SocketEndpoin
         LOG(MCdetailDebugInfo, unknownJob, "Action aborted before connecting to slave (%3d)", replyTag);
         return NULL;
     }
-
+#ifdef _CONTAINERIZED
+    // TODO: In K8s, this won't work in production as there's only a pool of NUM_SLAVE_CONNECT_PORT (20) ports for all ftslave instances
```

Comment:
it's okay, I see now I think.
childEp will always be localhost I think, so it runs ftslave as child process, then connects to it on localhost.

Reply:
Yes, it does work and the code from 173 is needed.  dfuserver spawns off local ftslave and then it connects to the ftslave via tcp port.  The retries are necessary because the port may have been spawned by a different dfuserver job.  My concern was related to the number of ports available to all the ftslaves - Gavin doesn't think it will be a problem. If another job's ftslave is using the port, the 20 retries will ensure that eventually this job's ftslave will succeed.... I will have another look into this and determine if NUM_SLAVE_CONNECT_PORT needs to be higher for containerized environment: https://track.hpccsystems.com/browse/HPCC-25591